tripy/tripy/frontend/trace/ops/resize.py

```diff
@@ -0,0 +1,152 @@
+#
+# SPDX-FileCopyrightText: Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from dataclasses import dataclass
+from typing import Sequence, Union
+
+from tripy import export, constraints
+from tripy.common.exception import raise_error
+from tripy.frontend import utils as frontend_utils
+from tripy.frontend.trace.ops import utils as op_utils
+from tripy.frontend.trace.ops.base import BaseTraceOp
+
+
+@dataclass(repr=False)
+class Resize(BaseTraceOp):
+
+    mode: str
+    scales: Sequence[float]
+    align_corners: bool
+
+    infer_shape_output_idxs = op_utils.ShapeOutputIdxPolicies.never_return_shape
+
+    def infer_rank(self):
+        self.outputs[0].rank = self.inputs[0].rank
+
+    def infer_dtypes(self):
+        self.outputs[0].dtype = self.inputs[0].dtype
+
+    def to_flat_ir(self, inputs, outputs):
+        from tripy.flat_ir.ops import ResizeCubicOp, ResizeLinearOp, ResizeNearestOp
+
+        if self.scales:
+            from tripy.common.datatype import float32, int32
+            from tripy.flat_ir.ops import ConstantOp, ConvertOp, MulOp
+            from tripy.flat_ir.tensor import FlatIRTensor
+
+            # construct output_shape using scales
+            out_shape = (inputs[0].rank,)
+            scales_tensor = FlatIRTensor.build(
+                shape=out_shape,
+                rank=1,
+                dtype=float32,
```

Comment:
Do scales always remain in fp32?

Reply:
Yes, `scales` is provided as static python numbers, it will be initialized as a fp32 tripy tensor.
Then output shape is calculated as `input.shape.cast(fp32) * scales`